### ISSUE TITLE
no-unused-vars raised when template literals

### ISSUE THREAD

**Issue by jeremychone** [eslint/eslint#5376]
Not sure if I am missing something but variables used template literals do not seem to count as used vars.

![image](https://cloud.githubusercontent.com/assets/74671/13242825/a6afc124-d9ad-11e5-84c5-7db1585f339c.png)

```
	proc.stdout.on('data', (data) => {
		console.log("stdout: `${data}`");
	});
```

Traditional javascript do not raise any warning. 
![image](https://cloud.githubusercontent.com/assets/74671/13242863/006a152a-d9ae-11e5-9724-924791ad0559.png)

Here is my .eslintrc.json

```json
{
    "env": {
        "browser": true,
        "commonjs": true,
        "es6": true,
        "node": true
    },
    "extends": "eslint:recommended",
    "parserOptions": {
        "sourceType": "module"
    },
    "rules": {
        "indent": [
            2,
            "tab"
        ],
        "linebreak-style": [
            2,
            "unix"
        ],
        "quotes": [
            0,
            "double"
        ],
        "semi": [
            2,
            "always"
        ], 
        "no-undef": [
        0],
        "no-console": [
        0], 
        "no-unused-vars": [
        1]
    }
}
```

**Comment by mysticatea** [comment]
Thank you for this issue.

```
"stdout: `${data}`"
```

This is a string, not a template.
Please fix it with:

```
`stdout: ${data}`
```

**Comment by jeremychone** [comment]
Ho, now, I am embarrassed, you are right, it passes now. 
![image](https://cloud.githubusercontent.com/assets/74671/13242906/85a37ab0-d9ae-11e5-8387-7061e0d62a58.png)

Silly and lazy copy/paste when playing around. Strange thing is that my wrong code ran on node.js 5.x. 

Btw, just moved to eslint from jshint and looks very good and smooth so far. Great job!

**Comment by mysticatea** [comment]
Thank you for using ESLint! :+1: 

Actually, backticks seem to be confusing people a bit.
I have seen reports about template literals sometimes.
